packages/lesswrong/server/recommendations.js

```diff
@@ -159,18 +160,78 @@ const getRecommendations = async ({count, algorithm, currentUser}) => {
   }
 };
 
+const getResumeSequences = async (currentUser, context) => {
+  if (!currentUser)
+    return [];
+  if (!currentUser.partiallyReadSequences)
+    return [];
+  
+  return Promise.all(_.map(currentUser.partiallyReadSequences,
+    async partiallyReadSequence => ({
+      sequence: partiallyReadSequence.sequenceId
+        ? await context["Sequences"].loader.load(partiallyReadSequence.sequenceId)
```

Comment:
I would probably deconstruct `partiallyReadSequence` before you go into this case. I expect this to make this a lot easier to understand.